pyro/distributions/diag_normal_mixture_shared_cov.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+import torch
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+from torch.distributions import Categorical, constraints
+
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.util import sum_leftmost
+
+
+class MixtureOfDiagNormalsSharedCovariance(TorchDistribution):
+    """Mixture of Normal distributions with diagonal covariance matrices. That is,
```

Comment:
nit: could you follow our standard docstring format:
```py
"""
First a one line description.

Then a paragraph of longer description that is wrapped at 80 characters
so that if fits into help() messages in Jupyter. This is easy to do in vim:
select the lines to wrap and type gq<ENTER>.

...
"""
```

Reply:
done